src/zarr/codecs/_v2.py

```diff
@@ -43,10 +44,16 @@ async def _decode_single(
 
         # view as numpy array with correct dtype
         chunk = ensure_ndarray_like(chunk)
+        print(chunk)
+        print(chunk.dtype)
         # special case object dtype, because incorrect handling can lead to
         # segfaults and other bad things happening
         if chunk_spec.dtype != object:
-            chunk = chunk.view(chunk_spec.dtype)
+            try:
+                chunk = chunk.view(chunk_spec.dtype)
+            except TypeError:
```

Comment:
If possible, could you add a comment explaining what this type error is catching? Is it perhaps compatibility for older versions of numpy?